docs/user-guide/collectWhileInState.md

```diff
@@ -0,0 +1,89 @@
+# collectWhileInState()
+
+This one is useful if you want to collect a `Flow` (from Kotlin Coroutines) only while being exactly in that state.
+To give a concrete example how this is useful let's extend our `ItemListStateMachine` example.
+Let's say whenever our state machine is in `Error` state we want
+to retry loading the items after 3 seconds in `Error` state or anytime before the 3 seconds have elapsed if the user clicks the retry button.
+Furthermore the 3 seconds countdown timer should be displayed in our UI as well.
+This is how the UI should look like:
+
+![Automatically Retry](../images/error-countdown.gif)
+
+To implement this let's first extend our `Error` state:
+
+```kotlin
+data class Error(
+    val cause: Throwable,
```

Comment:
extreme nit: I really dislike Throwables in state because they lead to all these technical error messages that a user should never see but some websites and apps show to users

Reply:
Agreed, it is also hard to test s data class with exceptions. Will change that example later tonight

Reply:
Updated with an error message instead of throwable. Was considering removing it completely, but then the docs would change from `object Error` to `data class Error` somewhen along the way (for the error countdown example) ... Maybe we change it later